### ISSUE TITLE
generated encrypted + licensed EPUB is returned as complete attachement (better to return link to allow client to performed optimized download?)

### ISSUE THREAD

**Issue by danielweck** [readium/readium-lcp-server#45]
https://github.com/readium/readium-lcp-server/blob/feature/newdb/lcpserver/api/license.go#L308

``` go
    ep.Add("META-INF/license.lcpl", &buf, uint64(buf.Len()))
    w.Header().Add("Content-Type", "application/epub+zip")
    w.Header().Add("Content-Disposition", fmt.Sprintf(`attachment; filename="%s"`, content.Location))
    ep.Write(w)
```

Problem if EPUB is very large. Better if client app has option to download in optimized manner, rather than `Content-Disposition`?.

**Comment by llemeurfr** [comment]
The codebase has evolved since then.

**Comment by danielweck** [comment]
I am not against closing this issue, as I do not have concrete metrics that demonstrate the (in)efficiency of creating an HTTP response body with the full encrypted/licensed EPUB (+ `content-disposition`) versus another transfer method (e.g. optimized storage + HTTP hosting solution). Furthermore, I believe that this functionality is not really used in production, as most clients download the LCP license on its own, then the encrypted publication (from a server optimised for distributing files), and the client takes care of the final `META-INF/license.lcpl` zip injection.

However I just want to point out that although the codebase has changed since this issue was created, the logic remains the same in `GenerateLicensedPublication()` and `GetLicensedPublication()`, respectively:

https://github.com/readium/readium-lcp-server/blob/c692a1744d151f292f3a60a96f4ffe1ec254e122/lcpserver/api/license.go#L449-L476

https://github.com/readium/readium-lcp-server/blob/c692a1744d151f292f3a60a96f4ffe1ec254e122/lcpserver/api/license.go#L370-L402

In other words, `buildLicencedPublication()` creates an in-memory buffer that contains the entire encrypted EPUB with its pre-injected `META-INF/license.lcpl` (and this buffer is subsequently fed into an HTTP response body).

https://github.com/readium/readium-lcp-server/blob/c692a1744d151f292f3a60a96f4ffe1ec254e122/lcpserver/api/license.go#L156-L192

This is inefficient for large files, but I believe this delivery method is not actually used by any LCP clients in practice (as I mentioned above).

**Comment by snemegeer** [comment]
We do insert the license in the epub and then send it to the client. The client receives an epub with the license in it.  This has an advantage for the user, and poses no problem for most (small) epub files.  The user receives an epub file he can open immediately.  
I think both methods should remain possible.
